gulpfile.mjs

```diff
@@ -11,29 +11,54 @@ import gulp from 'gulp';
 import serveDev from './dev-server/serve-dev.js';
 import servePackage from './dev-server/serve-package.js';
 import tailwindConfig from './tailwind.config.mjs';
+import annotatorTailwindConfig from './tailwind-annotator.config.mjs';
+import sidebarTailwindConfig from './tailwind-sidebar.config.mjs';
 
 gulp.task('build-js', () => buildJS('./rollup.config.mjs'));
 gulp.task('watch-js', () => watchJS('./rollup.config.mjs'));
 
-gulp.task('build-css', () =>
+gulp.task('build-annotator-tailwind-css', () =>
+  buildCSS(['./src/styles/annotator/annotator.scss'], {
+    tailwindConfig: annotatorTailwindConfig,
+  })
+);
+
+gulp.task('build-sidebar-tailwind-css', () =>
   buildCSS(
     [
-      // Hypothesis client
-      './src/styles/annotator/annotator.scss',
+      // sidebar styles (with tailwind)
+      './src/styles/sidebar/sidebar.scss',
+      // TODO: After tailwind migration complete, should this
+      // be managed with its own tailwind config?
+      './src/styles/ui-playground/ui-playground.scss',
+    ],
+    { tailwindConfig: sidebarTailwindConfig }
+  )
+);
+
+gulp.task('build-standalone-css', () =>
```

Comment:
These CSS files don't use any Tailwind classes that I can see. In the case of `highlights.scss`, I think it may be wise to _avoid_ Tailwind or add some additional scoping selectors to its rules (can we do that?) because this CSS is loaded directly into third-party web pages, and thus its styles may conflict with the pages. In the case of PDF.js, a conflict is still possible, but much more manageable because we know what the web app is.

Reply:
This situation has a few facets; let me outline what I understand and we can figure out the best way forward.

The way that `frontend-build`'s `sass` task works currently is that all stylesheets piped through the `sass` task are processed by tailwind if the `tailwind` package is installed, using the provided `tailwindConfig` or a fallback config object.[^1]

Similarly to how you can process a `.scss` file written in plain CSS with SASS (it's just a no-op), processing a source file with "nothing to do" for tailwind is also fine. Tailwind's not gonna mess with selectors. And if the selectors are not inside of a `@layer`, tailwind will always include them in output (it won't reference its `content` globs).

We have a few options available that we could compose:

* We could just copy the minified vendor CSS into place instead of processing it at all (that won't be applicable to all of these source styles).
* We could write a separate task that auto-prefixes, minifies and copies but does not process with SASS or tailwind. I'm not super keen on this but it's plausible.
* We could leave as-is but update the `sass` task in `frontend-build` such that we don't need to pass a (distracting) `tailwindConfig`) object[^1]. This would be my vote.
* We could adapt the `sass` task to be able to disable tailwind via an option.

[^1]: I don't want to distract from the main thrust of this conversation, but you may be wondering _why are we providing a `tailwindConfig` option for this set of stylesheet sources?_ (when tailwind isn't relevant to them). The answer is that there was a change to `tailwind` since the `frontend-build` task was updated, such that a configuration without a `content` property will trigger a warning on build. i.e. the current "default configuration" object in `frontend-build` is enough to keep tailwind from _crashing_, but it lacks a `content` value and needs to be updated to eliminate that warning. "It's on my list".

Reply:
> Tailwind's not gonna mess with selectors. 

Right. My thinking was more that someone in future might come along and try to replace the highlight classes with Tailwind and not realize why it isn't using it. A possible solution here might just be a comment in the gulpfile indicating that certain stylesheets should not use Tailwind and the reasons for that.

> We could leave as-is but update the sass task in frontend-build such that we don't need to pass a (distracting) tailwindConfig) object[1](https://github.com/hypothesis/client/pull/4514#user-content-fn-1-d46245ef650695d99d2aaf9a69dbca8f). This would be my vote.

This sounds good to me, but also we could leave the code as is but include a comment to note that these stylesheets shouldn't actually use Tailwind.

Reply:
Roger. I've got "update `sass` task in `frontend-build`" on my TODO list, such that we can stop sending a `tailwindConfig` here.